aclk/https_client.h

```diff
@@ -77,4 +80,36 @@ void https_req_response_init(https_req_response_t *res);
 
 int https_request(https_req_t *request, https_req_response_t *response);
 
+// we expose previously internal parser as this is usefull also from
+// other parts of the code
+enum http_parse_state {
+    HTTP_PARSE_INITIAL = 0,
+    HTTP_PARSE_HEADERS,
+    HTTP_PARSE_CONTENT
+};
+
+typedef uint32_t parse_ctx_flags_t;
+
+#define HTTP_PARSE_FLAG_DONT_WAIT_FOR_CONTENT ((parse_ctx_flags_t)0x01)
+
+#define HTTP_PARSE_FLAGS_DEFAULT ((parse_ctx_flags_t)0)
+
+typedef struct {
+    enum http_parse_state state;
+    int content_length;
+    int http_code;
+    c_rhash headers;
+    parse_ctx_flags_t flags;
+} http_parse_ctx;
+
+void http_parse_ctx_create(http_parse_ctx *ctx);
+void http_parse_ctx_destroy(http_parse_ctx *ctx);
+
+#define HTTP_PARSE_NEED_MORE_DATA  0
+#define HTTP_PARSE_SUCCESS   1
+#define HTTP_PARSE_ERROR    -1
```

Comment:
Considering the value we have for these three constant, I suggest to convert them to `enum`.

Reply:
converted